src/release_notes_workflow/release_notes.py

```diff
@@ -82,3 +88,56 @@ def check(self, component: InputComponentFromSource, build_version: str, build_q
                 else:
                     results.append(None)
         return results
+
+    def generate(self, component: InputComponentFromSource, build_version: str, build_qualifier: str, manifest_path: str = None) -> None:
+        """Generate AI-powered release notes for a component."""
+        with TemporaryDirectory(chdir=True) as work_dir:
+            # Initialize processor with initial date
+            processor = Processor(build_version, self.date)
+            
+            # First, get the date of the last tag in the opensearch-build repository
+            baseline_date = self.date
+            try:
+                # Check out the opensearch-build repository to get the last tag date
+                with GitRepository(
+                        "https://github.com/opensearch-project/opensearch-build.git",
+                        "main",
+                        os.path.join(work_dir.name, "opensearch-build"),
+                        fetch_depth=0  # Fetch full history to get tags
+                ) as build_repo:
+                    baseline_date = processor.get_last_tag_date(build_repo, self.date)
+                    # Update processor with the final baseline date
+                    processor = Processor(build_version, baseline_date)
+            except Exception as e:
+                logging.warning(f"Failed to get last tag date from opensearch-build: {e}")
+            
+            with GitRepository(
+                    component.repository,
+                    component.ref,
+                    os.path.join(work_dir.name, component.name),
+                    component.working_directory,
+                    fetch_depth=0  # Fetch full history to get all commits
+            ) as repo:
+                # Initialize AI generator
+                ai_generator = AIReleaseNotesGenerator(
+                    github_token=None,
+                    version=build_version,
+                    baseline_date=baseline_date,
+                    test_mode=self.test_mode
+                )
+
+                release_notes = ReleaseNotesComponents.from_component(component, build_version, build_qualifier, repo.dir)
+                
+                # Try to fetch CHANGELOG.md directly from GitHub
+                content = processor.fetch_changelog_from_github(component)
```

Comment:
This can be further simplified to below: 
`changelog_exist = os.path.isfile(os.path.join(repo.dir, 'CHANGELOG.md'))`.